File: backend/schemas/daily.py
"""
Daily summary schemas
"""
from pydantic import BaseModel
from uuid import UUID
from datetime import date
from typing import Optional

from schemas.health import HealthEntryResponse
from schemas.college import CollegeTaskResponse
from schemas.journal import JournalEntryResponse


class DailySummaryResponse(BaseModel):
    """Daily summary response"""
    date: date
    user_id: UUID
    health: Optional[HealthEntryResponse] = None
    college_tasks: list[CollegeTaskResponse] = []
    journal_entries: list[JournalEntryResponse] = []
    total_expenses: float = 0.0
    water_intake: int = 0
    steps: int = 0
    tasks_completed: int = 0
    tasks_pending: int = 0


class SendDailySummaryEmailRequest(BaseModel):
    """Send daily summary email request"""
    date: Optional[date] = None  # Defaults to today
    recipient_email: Optional[str] = None  # Defaults to user's email

File: src/api/daily.ts
/**
 * Daily Summary API Service
 */
import { apiClient } from './client';
import { HealthEntry } from './health';
import { CollegeTask } from './college';
import { JournalEntry } from './journal';

export interface DailySummary {
  date: string;
  user_id: string;
  health?: HealthEntry;
  college_tasks: CollegeTask[];
  journal_entries: JournalEntry[];
  total_expenses: number;
  water_intake: number;
  steps: number;
  tasks_completed: number;
  tasks_pending: number;
}

export const dailyApi = {
  /**
   * Get daily summary for a specific date
   */
  getSummary: async (date?: string): Promise<DailySummary> => {
    const response = await apiClient.get<DailySummary>('/daily/summary', {
      params: date ? { date } : undefined,
    });
    return response.data;
  },

  /**
   * Send daily summary email
   */
  sendEmail: async (data?: { date?: string; recipient_email?: string }): Promise<void> => {
    await apiClient.post('/daily/email', data);
  },
};
